packages/transformer/src/IModelTransformer.ts

```diff
@@ -411,18 +414,144 @@ export class IModelTransformer extends IModelExportHandler {
 
   private _isSynchronization = false;
 
-  private get _isReverseSynchronization() {
-    return this._isSynchronization && this._options.isReverseSynchronization;
+  /**
+   * A private variable meant to be set by tests which have an outdated way of setting up transforms. In all synchronizations today we expect to find an ESA in the branch db which describes the master -> branch relationship.
+   * The exception to this is the first transform aka the provenance initializing transform which requires that the master imodel and the branch imodel are identical at the time of provenance initialization.
+   * A couple ofoutdated tests run their first transform providing a source and targetdb that are slightly different which is no longer supported. In order to not remove these tests which are still providing value
+   * this private property on the IModelTransformer exists.
+   */
+  private _allowNoScopingESA = false;
+
+  public static noEsaSyncDirectionErrorMessage =
+    "Couldn't find an external source aspect to determine sync direction. This often means that the master->branch relationship has not been established. Consider running the transformer with wasSourceIModelCopiedToTarget set to true.";
+
+  /**
+   * querys the scopes external source and sets aspectId, version and jsonProperties on the provided aspectProps if available.
+   * @param dbToQuery db to run the query on for scope external source
+   * @param aspectProps aspectProps to search for
+   * @param opts opts containing preserveVersionReceived and getJsonProperties. if preserveVersionReceived is true and an aspect is found with the provided aspectProps,
+   * then the version passed as part of the aspectProps will be kept on the aspectProps.
+   * If false or not provided, the version on the aspectProps will be overwritten with the version from the queried aspect.
+   */
+  public static queryScopeExternalSource(
```

Comment:
Rename from `queryScopeExternalSource` to `queryTargetScopeExternalSourceAspect`?

Also can we always query the JSON properties and just ignore them in callers that don't need them? I don't remember why I differentiated it last time I refactored however... if we find there is a reason, let's document it over the `getJsonProperties` option

Reply:
`queryTargetScopeExternalSourceAspect`. Is TargetScopeESA different than the scoping ESA? I suspect it is, but just asking since this function is used for more than just the scoping ESA

Reply:
yes I think I meant to go back and edit this comment before I posted my review. Let's just drop `Target` and do `queryScopeExternalSourceAspect`? I wanted to add "Target" to align better with the concept of `targetScopeElement`, which is always in the branch (hence "target", but in a reverse sync the branch (also provenance db) is the source)
But I think you're right we use it in other places that makes that less of a good idea

Reply:
Regarding `getJsonProperties`, all tests still pass if we are to remove the `getJsonProperties` option and just always grab the jsonProperties. I guess my only concern is can we be sure that every ESA has jsonProperties? Previously this function was private and is now public static, so I think in theory anyone could look for any type of ESA, and jsonProperties are not required to exist on ESA

Reply:
Added check for isNull before getting the jsonprops.